### PR TITLE
Feature/update financial transaction category controller #257

### PR DESCRIPTION
**Pull request by sylvvester**
#### Checklist template for LEADERS

&#x2611; | &#x2610;

&#x2611; New code is covered with tests and coverage requirements are met.
&#x2610; Build is not failing.
&#x2610; All tests are passing.
&#x2610; All conflicts are resolved and the branch is rebased on top of the up-to-date parent branch.
&#x2611; All issue requirements are implemented.
&#x2610; At least two reviewers are assigned.
&#x2610; The Pull request is linked to the related issue.
&#x2610; Correct branch name. Convention [https://github.com/bycza-zagroda/track-expenses-app-documentation/blob/develop/githubwork/GITHUB_WORK.md]
&#x2610; Correct unit test name - suffix 'Test' in the end.
&#x2610; Correct integration test name - suffix 'IT' in the end.
&#x2610; Positive and negative test paths have been implemented.

### CODE REVIEW

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryController.java [hunk 1]
```diff
@@ -32,38 +33,59 @@ public class FinancialTransactionCategoryController {
...
     @GetMapping()
-    ResponseEntity<List<FinancialTransactionCategoryDTO>> getFinancialTransactionCategories(){
-        List<FinancialTransactionCategoryDTO> financialTransactionCategoryDTOList = financialTransactionCategoryService.getFinancialTransactionCategories();
+    ResponseEntity<List<FinancialTransactionCategoryDTO>> getFinancialTransactionCategories(Principal principal) {
+
+        Long userId = Long.valueOf(principal.getName());
+
+        List<FinancialTransactionCategoryDTO> financialTransactionCategoryDTOList
+                = financialTransactionCategoryService.getFinancialTransactionCategories(userId);
         return new ResponseEntity<>(financialTransactionCategoryDTOList, HttpStatus.OK);
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryController.java:55]: One empty line before this one please - just like you got it in L:44

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryController.java [hunk 2]
```diff
@@ -32,38 +33,59 @@ public class FinancialTransactionCategoryController {
...
-            @RequestBody FinancialTransactionCategoryCreateDTO financialTransactionCategoryCreateDTO) {
+            @RequestBody FinancialTransactionCategoryCreateDTO financialTransactionCategoryCreateDTO,
+            Principal principal) {
+
+        Long userId = Long.valueOf(principal.getName());
+
         FinancialTransactionCategoryDTO financialTransactionCategoryDTO =
-                financialTransactionCategoryService.createFinancialTransactionCategory(financialTransactionCategoryCreateDTO);
+                financialTransactionCategoryService.createFinancialTransactionCategory(financialTransactionCategoryCreateDTO, userId);
         return new ResponseEntity<>(financialTransactionCategoryDTO, HttpStatus.CREATED);
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryController.java:67]: One empty line before this one please - just like you got it in L:44

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryController.java [hunk 3]
```diff
@@ -32,38 +33,59 @@ public class FinancialTransactionCategoryController {
...
+            @Min(1) @NotNull @PathVariable(name = "id") Long categoryId,
+            @Valid @RequestBody FinancialTransactionCategoryUpdateDTO financialTransactionCategoryUpdateDTO,
+            Principal principal) {
+
+        Long userId = Long.valueOf(principal.getName());
+
         FinancialTransactionCategoryDTO financialTransactionCategoryDTO
-                = financialTransactionCategoryService.updateFinancialTransactionCategory(id, financialTransactionCategoryUpdateDTO);
+                = financialTransactionCategoryService.updateFinancialTransactionCategory(categoryId, userId, financialTransactionCategoryUpdateDTO);
         return new ResponseEntity<>(financialTransactionCategoryDTO, HttpStatus.OK);
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryController.java:80]: One empty line before this one please - just like you got it in L:44

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryController.java [hunk 4]
```diff
@@ -32,38 +33,59 @@ public class FinancialTransactionCategoryController {
...
 
     @DeleteMapping("/{id}")
-    public ResponseEntity<Void> deleteFinancialTransactionCategoryById(@Min(1) @NotNull @PathVariable Long id) {
-        financialTransactionCategoryService.deleteFinancialTransactionCategory(id);
+    public ResponseEntity<Void> deleteFinancialTransactionCategoryById(@Min(1) @NotNull @PathVariable(name = "id")  Long categoryId, Principal principal) {
+
+        Long userId = Long.valueOf(principal.getName());
+
+        financialTransactionCategoryService.deleteFinancialTransactionCategory(categoryId, userId);
         return new ResponseEntity<>(HttpStatus.NO_CONTENT);
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryController.java:89]: One empty line before this one please - just like you got it in L:44

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Request changes

**Review by dawciobiel — DISMISSED** [review]
LGTM

**Review by dawciobiel — APPROVED** [review]
LGTM

**Review by mirekgab — APPROVED** [review]

**Pull request merged**